spec/lib/miq_expression_spec.rb

```diff
@@ -17,31 +17,62 @@
       expect(sql).to eq("\"vms\".\"name\" = 'foo'")
     end
 
+    it "generates the SQL for a = expression with expression as a value" do
+      sql, * = MiqExpression.new("=" => {"field" => "Vm-name", "value" => "Vm-name"}).to_sql
+      expect(sql).to eq("\"vms\".\"name\" = \"vms\".\"name\"")
```

Comment:
Can you try to execute it? I think it will fail. It would be good if you could add some more integrated tests for these to prove that it has the right behavior. See [this](https://github.com/ManageIQ/manageiq/blob/master/spec/lib/miq_expression_spec.rb#L367) block for some examples.